contract/contracts/Subscriptions.sol

```diff
@@ -99,40 +87,53 @@ contract Subscriptions {
     // and cannot be recovered by the user.
     // Defined as `rate * max(0, min(block, end) - start)`
     function locked(Subscription storage sub) private view returns (uint128) {
-        uint64 currentBlock = uint64(block.number);
-        int256 len = Prelude.max(
-            0,
-            Prelude.min(int64(currentBlock), int64(sub.end)) - int64(sub.start)
+        uint256 len = uint256(
+            SignedMath.max(
+                0,
+                int256(Math.min(block.number, sub.end)) - int64(sub.start)
+            )
         );
-        return sub.rate * uint128(uint256(len));
+        return sub.rate * uint128(len);
     }
```

Comment:
Another think I just thought of, if we make this public then you could also change it so instead of taking a struct as parameter it takes the struct members. 

```
// Instead of this
function locked(Subscription storage sub)

// This
function locked(uint64 start, uint64 end, uint128 rate)
```

The only reason i suggest this is because it makes it easier to interact with the contract via Etherscan for example. So it's just a suggestion, I haven't checked how it changes the rest of the code maybe it's not worth it (or maybe you think it's a silly thing haha)

Reply:
If we don't want to make it public then we should remove the function as it's not being used.

Reply:
I think exposing it as a public function to let a user see how many locked tokens they have makes sense. I wonder if we should lock it down to the user though.

Reply:
Exposed the function, updated the params to break out of the Subscription object. Also added a separate to find the locked tokens for a user: [5b5324c](https://github.com/edgeandnode/subscription-payments-contract/pull/9/commits/5b5324c91ce50b9c915371be44c0ce35da3928cc)
@tmigone should this be locked down so only a user can see their locked tokens?

Reply:
nice! I dont think locking down is necessary, it's all public data anyways and the user address it's not something that can be easily guessed or associated with anyone.